src/map/Map.cpp

```diff
@@ -0,0 +1,239 @@
+ */
+int Map::loadFileData() {
+    // Open file stream for selected map file
+    ifstream file;
+    file.open(fname.c_str());
+
+    if (file.fail()) {
+        printf("failed to open file\n");
+        return 0;
+    }
+
+    // Array of wallstart positions
+    struct MapPoint wallStart[MAX_WALLS];
+    // Indices
+    int j = 0;
+    int i = 0;
+
+    // Character Variable
+    char ch;
+    while (file >> noskipws >> ch) {
+        switch (ch) {
+            case ',':
+                continue;
+            case '\n':
+                j = 0;
+                i++;
+                continue;
+            case WALL:              //Wall piece found, update AI Map
+                AIMap[i][j] = 1;
+                break;
+            case WALL_START:        //Label start of wall rectangle
+                wallStart[ws].x = j;
+                wallStart[ws].y = i;
+                ws++;
+                AIMap[i][j] = 1;
+                break;
+            // case CONCRETE_START:    //Start of concrete area
+            case ZOMBIE_SPAWN:      //Zombie spawn Points
+                zombieSpawn[zs].x = j;
+                zombieSpawn[zs].y = i;
+                zs++;
+                break;
+            case SHOP_SPOT:         // Shop spot points
+                shops[s].x = j;
+                shops[s].y = i;
+                s++;
+                break;
+            case BASE_START:        //Start of base area
+                base.x = j;
+                base.y = i;
+                break;
+        }
+        mapdata[i][j] = ch;               //Store map file value
+        j++;
+    }
+    file.close();
+
+    genWalls(wallStart);
+
+    // printData();
+    return 1;
+}
+
+/**
+ * Date: Mar. 23, 2017
+ * Author: Jordan Lin
+ * Function Interface: void genWalls(const struct MapPoint wstart[MAX_WALLS])
+ * Parameters:
+ *      wstart - Array of wall start MapPoints.
+ * Description:
+ * Read's the Map CSV file map data with the appropriate tile IDs
+ */
+void Map::genWalls(const struct MapPoint wstart[MAX_WALLS]) {
+    for (int q = 0; q < ws; q++) {
+        int startx = wstart[q].x;
+        int starty = wstart[q].y;
+        int nextx = startx + 1;
+        int nexty = starty + 1;
+        int endx;
+        int endy;
+        while (mapdata[starty][nextx] == WALL) {
+            nextx ++;
+        }
+        endx = nextx;
+        while (mapdata[nexty][startx] == WALL) {
+            nexty++;
+        }
+        endy = nexty;
+        MapWall temp;
+        temp.x = startx * T_SIZE;
+        temp.y = starty * T_SIZE;
+        temp.width = (endx - startx) * T_SIZE;
+        temp.height = (endy - starty) * T_SIZE;
+        walls.push_back(temp);
+    }
+}
+
+/**
+ * Date: Mar. 23, 2017
+ * Author: Jordan Lin
+ * Function Interface: void mapLoadToGame()
+ * Description:
+ * Creates the wall structures using the GameManager createWall function.
+ */
+void Map::mapLoadToGame() {
+    for(auto w : walls) {
```

Comment:
[Did you even read the implementation guide?](https://github.com/bcit-datacomm/4981_Linux_Game/wiki/Implementation-Guide#use-of-const-by-default-is-required)